### ISSUE TITLE
Question: Strategy for packaging styles

### ISSUE THREAD

**Issue by bmatto** [react-toolbox/react-toolbox#1257]
This is more of a question than an issue, hopefully someone has an answer for me.

I am trying to publish a react component lib that has some styles associated with it.  I would like that a component when imported also bundles the theme information with it.  Like react-toolbox I am using scss and react-css-themr.

I noticed that in the build setup for this project, the src is run through babel with 'lib' as the output, then the scss is moved via cpx.  I have attempted to recreate this approach with my codebase.  Here are the package.json scripts

```json
"scripts": {
    "test": "jest",
    "test:watch": "yarn test -- --watch",
    "babel": "babel ./src -d lib",
    "build": "NODE_ENV=production yarn babel && yarn sass",
    "prebuild": "rimraf lib",
    "prepublish": "yarn test && yarn prebuild && yarn build",
    "sass": "cpx \"./src/**/*.scss\" ./lib"
  }
```

The build process works out happily.  I am testing this locally by using npm link.  When importing this into another application in the system it falls over complaining that the "postcss" webpack loader can't be found.  True, that the application doing the importing is using postcss to handle scss and it does so without issue for other assets including those coming from react-toolbox.

TLDR - How are you folks publishing this library such that other applications are able to import the modules with the scss files bundled.

Thank you so much for this lib, and any response! Cheers.

**Comment by javivelasco** [comment]
Hi @bmatto! 

Sadly my answer is: don't do that. The main problem with this library is that you have a lot to setup to get it working. As you mentioned, when the component you are defining bundles the styles, the consumer bundler should be configured to use css modules and, not only that, it should resolve into the object. I mean, you don't only need css loader but modules too.

A lot of people often open issues related to their webpack configuration so my first attempt to solve it was to extract the CSS import from the component definition. This is why if you browse the code you'd see components are actually factories. They are invoked and exposed without styles and, in the `index.js` file of each component folder the CSS is required and injected. This means that:

- If you want to import bundling CSS you import from index.
- If you are going to inject the module using themr or anything else, import from the component.

It is tricky and a source of confusion and maybe it'd be better to implement this in two different packages but it is like it is now and it works.

With 2.0 version we migrated everything to PostCSS and styles are processed so you only need to provide custom variables. This is ok, but still needs configuration. To overcome this I developed [react-toolbox-themr]() which is an extractor of css and a provider that helps you skipping configuration. It has an important issue tho: the dependency resolution is broken and you have to programmatically specify what will be imported.

After thinking a lot about it, the right solution IMO is to use some CSS in JS solution but to avoid marrying with an specific one I'm working on an approach of agnostic components. Then we can use styled components as the default styling solutions and to forget about configuration. You just require a component and it works.

Please ask if you need, hope it helps

**Comment by bmatto** [comment]
Thanks for this response btw, and great work on the lib :)
